### PR TITLE
Feature: batchnorm gpu test

### LINKED ISSUE #4: Add tests for Batch Normalization operator

### PR DESCRIPTION
**Pull request by harz05**
Closes #4

This implements the 3 GPU/Alpaka methods for `ROperator_BatchNormalization` and adds a test in the Alpaka CUDA test suite.

### `ROperator_BatchNormalization.hxx`

- `Generate_GPU_Kernel_ALPAKA` : elementwise kernel struct. `Initialize()` already pre-expands per-channel scale/bias/mean tensors to the full `[N,C,H,W]` shape and folds `1/sqrt(var + eps)` into scale, so the per-element formula reduces to `y[i] = (x[i] - mean[i]) * scale[i] + bias[i]`. Kernel pattern is same like other operators
- `Generate_GPU_Kernel_Definitions_ALPAKA` : declares the kernel instance used at launch.
- `Generate_GPU_ALPAKA` : sets up work division via alpaka::getValidWorkDiv and dispatches the kernel with alpaka::exec. Uses the `deviceBuf_<name>` buffers for all weight tensors consistent with how `RModel_ALPAKA.cxx` allocates them.

### Files for testing:

- `BatchNormModelGenerator.py` : creates `BatchNorm.onnx` from `nn.BatchNorm2d(2)` in eval mode with fixed weights. Input shape `(1, 2, 2, 2)`, scale `[1, 2]`, bias `[0, 0.5]`, running mean `[0.5, 3]`, running var `[1, 4]`. Also prints the reference floats used in the `.ref.hxx`.
- `input_models/BatchNorm.onnx` : the exported ONNX model (opset 13).
- `input_models/references/BatchNorm.ref.hxx` : 8 reference output values computed by PyTorch.
- `TestCustomModelsFromONNXForAlpakaCuda.cxx` : added `SofieAlpakaTest.BatchNorm` which loads the model on GPU, runs inference, and checks each output element against the reference within tolerance.

## Testing:

Tested on Google Colab (NVIDIA T4, CUDA 12.x). New test passes and all 26 tests in the suite pass with no regressions.

image

### CODE REVIEW

File: src/SOFIE_core/test/BatchNormModelGenerator.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+"""Generate BatchNorm.onnx and print reference output values for BatchNorm.ref.hxx.
+
+Model: nn.BatchNorm2d(2) in eval mode, input shape (1, 2, 2, 2).
+"""
+
```
Comment (sanjibansg) [inline src/SOFIE_core/test/BatchNormModelGenerator.py:6]: How will this file be executed? Also, since you provide the `.onnx` file anyway, how will this script be useful?

**Review by sanjibansg — COMMENTED** [review]
Hi @harz05,
Thanks for the PR. A question regarding the onnx file for testing.

File: src/SOFIE_core/test/BatchNormModelGenerator.py (hunk 1 above)
Comment (harz05) [reply src/SOFIE_core/test/BatchNormModelGenerator.py:6]: Thank you for reviewing,
this file is standalone and I included it for reproducibility so that the model and ref values can be regenerated and verified independently. `python3 BatchNormModelGenerator.py` executes the file and we get the .onnx along with the ref values.

**Comment by harz05** [comment]
closing this since upstream gpu/alpaka already has a new implementation of batchnorm test
However the current implementation has a scope for new tests which can be raised in a diferent PR.
